packages/block-editor/src/components/provider/with-registry-provider.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { withRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { storeConfig } from '../../store';
+import applyMiddlewares from '../../store/middlewares';
+
+const withRegistryProvider = createHigherOrderComponent( ( WrappedComponent ) => {
+	return withRegistry( ( { useSubRegistry = true, registry, ...props } ) => {
+		if ( ! useSubRegistry ) {
+			return <WrappedComponent registry={ registry } { ...props } />;
+		}
+
+		const [ subRegistry, updateRegistry ] = useState( null );
```

Comment:
Do we want to use "update" as the conventional vocabulary? From the [React documentation](https://reactjs.org/docs/hooks-intro.html), it seems conventional to encourage `[ foo, setFoo ]`.

Likewise, I'd expect:

```suggestion
		const [ subRegistry, setSubRegistry ] = useState( null );
```

Small detail, but these early introductions are likely to proliferate as setting some standard for how we use hooks.

Reply:
I agree with changing to `setSubRegistry` to follow the convention seemingly recommended in docs.

Reply:
Yeah, I like `set` thanks for the suggestion.